### PR TITLE
[FIXED JENKINS-41244] Be defensive when trying to set the browser

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-41244](https://issues.jenkins-ci.org/browse/JENKINS-41244)

@reviewbybees

### CODE REVIEW

**Comment by stephenc** [comment]
@svanoort @andresrc any chance you could re-review I believe I had the time comparison the wrong way around "to check how good the reviewers were (cough) (cough)"... yes that sounds like a legitimate excuse for my obvious mistake

**Comment by svanoort** [comment]
re-🐝  with some embarrassment...

**Comment by stephenc** [comment]
@svanoort I'm embarrassed that I didn't spot it when I wrote the code...

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java [hunk 1]
```diff
@@ -846,20 +853,43 @@ public SCM build(SCMHead head, SCMRevision revision) {
...
+    private void setBrowser(GitSCM scm, String repoUrl) {
+        try {
+            scm.setBrowser(new GithubWeb(repoUrl));
+        } catch (NoSuchMethodError e) {
+            Level level;
+            long now = System.currentTimeMillis();
+            long next = jenkins41244Warning.get();
+            if (now >= next) {
+                long newNext = now + TimeUnit.MINUTES.toMillis(5);
+                if (jenkins41244Warning.compareAndSet(next, newNext)) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]: Better to leave it to log handlers to decide how to deal with duplicated warnings.

**Pull request merged**